draft-packages/form/KaizenDraft/Form/Primitives/ToggleSwitch/ToggleSwitch.spec.tsx

```diff
@@ -0,0 +1,38 @@
+import { cleanup, fireEvent, render, screen } from "@testing-library/react"
+import * as React from "react"
+import { ToggledStatus, ToggleSwitchProps } from "./ToggleSwitch"
+import { ToggleSwitch } from "."
+import "@testing-library/jest-dom"
+
+afterEach(cleanup)
+
+const defaultToggleSwitchProps = {
+  id: "someToggleSwitchId",
+  onToggle: jest.fn(),
+  toggledStatus: ToggledStatus.OFF,
+}
+
+const renderToggleSwitch = (props?: ToggleSwitchProps) => {
+  const mergedToggleSwitchProps = { ...defaultToggleSwitchProps, ...props }
+
+  return render(<ToggleSwitch {...mergedToggleSwitchProps} />)
+}
+
+describe("<ToggleSwitch />", () => {
+  it("should call onToggle when toggle is changed", () => {
+    renderToggleSwitch()
+
+    fireEvent.click(screen.getByRole("checkbox"))
+    expect(defaultToggleSwitchProps.onToggle).toHaveBeenCalledTimes(1)
+  })
+
+  it("should show toggledStatus when passed through", async () => {
+    renderToggleSwitch()
+    expect(defaultToggleSwitchProps.toggledStatus).toEqual("off")
+  })
+
+  it("has disabled attribute when disabled prop passed in", async () => {
+    const { container } = renderToggleSwitch({ disabled: true })
+    expect(container.querySelector("[disabled]")).toBeTruthy()
```

Comment:
I don't mind this too much, but a way to make this more user behaviour focused could be to use the same code as you have in the `onToggle` test above, and make sure that clicking this disabled one _doesn't_ fire the `onToggle` method.

Reply:
Yep I tried that but unfortunately our logic to handle if it's disabled etc. is _inside_ of the `onToggle` so it still gets called, then decides what to do. So this would be a good case again for the `userEvent` library and potentially mocking the function.

Reply:
Oh right, fair enough 👍